swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/discovery/DiscoveryActivated.kt

```diff
@@ -0,0 +1,30 @@
+package com.netflix.spinnaker.swabbie.discovery
+
+import com.netflix.appinfo.InstanceInfo
+import com.netflix.spinnaker.kork.eureka.RemoteStatusChangedEvent
+import org.slf4j.LoggerFactory
+import org.springframework.context.ApplicationListener
+
+/**
+ * A component that starts doing something when the instance is up in discovery
+ * and stops doing that thing when it goes down.
+ */
+interface DiscoveryActivated : ApplicationListener<RemoteStatusChangedEvent> {
+
+  val onDiscoveryUpCallback: (event: RemoteStatusChangedEvent) -> Unit
+  val onDiscoveryDownCallback: (event: RemoteStatusChangedEvent) -> Unit
+
+  override fun onApplicationEvent(event: RemoteStatusChangedEvent) {
+    if (event.source.status == InstanceInfo.InstanceStatus.UP) {
+      log.info("Instance is {}... {} starting", event.source.status, DiscoveryActivated::class.java.simpleName)
```

Comment:
`getClass().simpleName` or whatever instead so the log messages use the actual class rather than always saying `DiscoveryActivated`